packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1824,9 +1824,106 @@ function scheduleWorkToRoot(fiber: Fiber, expirationTime): FiberRoot | null {
   return root;
 }
 
-export function warnIfNotCurrentlyBatchingInDev(fiber: Fiber): void {
+// in a test-like environment, we want to warn if dispatchAction() is
+// called outside of a TestUtils.act(...)/batchedUpdates/render call.
+// so we have a a step counter for when we descend/ascend from
+// actedUpdates() calls, and test on it for when to warn
+let actingUpdatesScopeDepth = 0;
+
+// we 'wait' for queued microtasks to resolve (after flushing effects)
+// using either setImmediate when available, or postMessage
+const nextTick =
```

Comment:
Nit: `nextTick` might not be the best tick because in Node it behaves like a microtask, not a macrotask. But you specifically want this to resolve after a macrotask.

Reply:
how does enqueueTask sound?